File: Java/Skracanie identyfikatorow.java
import java.util.Scanner;

public class Main 
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);
        int n = input.nextInt();
        String variableName = input.next();
        if (variableName.length() <= n)
            System.out.println(variableName);
        StringBuilder filteredName = new StringBuilder();
        for (char ch : variableName.toCharArray())
            if (Character.isLetterOrDigit(ch) || ch == '$')
                filteredName.append(ch);
        if (filteredName.length() <= n)
            System.out.println(filteredName.toString());
        StringBuilder noDigits = new StringBuilder();
        for (char ch : filteredName.toString().toCharArray())
            if (!Character.isDigit(ch))
                noDigits.append(ch);
        if (noDigits.length() <= n)
            System.out.println(noDigits.toString());
        String vowels = "AEIOUaeiou";
        boolean firstVowelFound = false;
        StringBuilder finalName = new StringBuilder();
        for (char ch : noDigits.toString().toCharArray()) 
        {
            if (vowels.indexOf(ch) != -1) 
            {
                if (!firstVowelFound) 
                {
                    firstVowelFound = true;
                    finalName.append(ch);
                }
            } 
            else
                finalName.append(ch);
        }
        if (finalName.length() <= n)
            System.out.println(finalName.toString());
        System.out.println(finalName.substring(0, n));
        input.close();
    }
}

File: C++/Skracanie identyfikatorow.cpp
#include <string>
#include <cctype>
#include <iostream>

bool isVowel(char c) 
{
    c = std::tolower(c);
    return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u';
}

int main() 
{
    int n;
    std::string variable_name;
    std::cin >> n;
    std::cin >> variable_name;
    if (variable_name.size() <= n)
        std::cout << variable_name << '\n';
    std::string filtered_name;
    for (char c : variable_name)
        if (std::isalnum(c) || c == '$')
            filtered_name += c;
    if (filtered_name.size() <= n)
        std::cout << filtered_name << '\n';
    std::string no_digits;
    for (char c : filtered_name)
        if (!std::isdigit(c))
            no_digits += c;
    if (no_digits.size() <= n)
        std::cout << no_digits << '\n';
    std::string final_name;
    bool first_vowel_found = false;
    for (char c : no_digits) 
    {
        if (isVowel(c)) 
        {
            if (!first_vowel_found) 
            {
                first_vowel_found = true;
                final_name += c;
            }
        }
        else
            final_name += c;
    }
    if (final_name.size() <= n)
        std::cout << final_name << '\n';
    std::cout << final_name.substr(0, n) << '\n';
    return 0;
}
